tests/test_signal_generator.cpp

```diff
@@ -0,0 +1,169 @@
+#include <hyro/SignalGeneratorComponent.h>
+#include <hyro/DigitalConverterComponent.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+
+#include <hyro/test/ComponentTester.h>
+
+#include <gtest/gtest.h>
+
+#define TEST_AMPLITUDE 100.0
+#define TEST_FREQUENCY 30.0
+#define TEST_COSINE false
+#define TEST_TIMESTAMP 1000000000
+#define TEST_FRAMID     "Teste 1"
+#define TEST_VALUE      20.0
+#define TEST_RETURNED_VALUE_POS 1.0
+#define TEST_RETURNED_VALUE_NEG 0.0
+
+static void
+__assert_and_run_state_machine(hyro::StateMachine &sm,
+                               const hyro::ComponentConfiguration &conf,
+                               const hyro::ConnectionConfiguration &endpoint)
+{
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.init(conf));
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.start());
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.connect(endpoint));
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.check());
+}
+
+using namespace signal_generator;
+using namespace digital_converter;
+
+namespace hyro
+{
+namespace test
+{
+
+TEST(ExampleTest, SignalGeneratorComponentCheck)
+{
+  //Configuration
+  auto generator_configuration = "{"
+                                 "outputs: {"
+                                 //"analog_output: { protocol: 'api', ip: '0.0.0.0',   port: '50051' },"
+                                 "analog_output: { protocol: 'api'},"
+                                 "fix_dynamic: { protocol: 'api'},"
+                                 "}"
+                                 "}";
+
+  //IntGeneratorComponent state machine
+  StateMachine sm(std::make_shared<SignalGeneratorComponent>("/generator"_uri));
+
+  auto result = sm.init(ComponentConfiguration(generator_configuration));
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  result = sm.start();
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  result = sm.connect(ConnectionConfiguration());
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  result = sm.check();
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  //Fake input channels
+  auto AnalogOut = std::make_shared<FakeInput<SignalMsgs>>("AnalogOut"_uri, "api", "/generator/analog_output");
```

Comment:
@rfcarneiro do not declare variables with the first letter capitalized. Change to `analog_out` or `analogOut`.